components/brave_new_tab_ui/reducers/new_tab_reducer.ts

```diff
@@ -179,6 +179,17 @@ export const newTabReducer: Reducer<NewTab.State | undefined> = (state: NewTab.S
       }
       break
 
+    case types.UPDATE_CLOCK_WIDGET: {
+      const { showClockWidget, clockFormat } = payload
+      preferencesAPI.saveShowClock(showClockWidget)
+      preferencesAPI.saveClockFormat(clockFormat)
+      const newState = {
+        ...state
+      }
+      newState.showClock = showClockWidget
```

Comment:
I don't think we actually need this? It should be almost instantaneously covered by the redux action which fires from the event handler when the preferences are updated. At least that's how all the other preferences are sent to the redux store as far as I remember.

Reply:
we definitely don't - I didn't actually even finish this 😛 It was setting `newState` but then never assigning `state` (whoops!). Removed